source/dxgi/dxgi_impl_display.hpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2024 Patrick Mours
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#pragma once
+
+#include <dxgi1_6.h>
+#include <xstring>
+#include "com_ptr.hpp"
+#include "reshade_api_object_impl.hpp"
+#include "reshade_api_display.hpp"
+
+namespace reshade::dxgi
+{
+	class display_impl : public api::api_object_impl<IDXGIOutput6 *, api::display>
+	{
+	public:
+		explicit display_impl(IDXGIOutput6 *output, DISPLAYCONFIG_PATH_TARGET_INFO *target);
+
+		/// <summary>
+		/// Indicates if cached properties are valid or potentially stale (i.e. refresh rate or resolution have changed).
+		/// </summary>
+		/// <remarks>
+		/// If this returns false, wait one frame and call runtime::get_active_display() to get updated data.
+		/// </remarks>
+		virtual bool is_current() const final;
+
+		/// <summary>
+		/// Gets the handle of the monitor this display encapsulates.
+		/// </summary>
+		virtual monitor get_monitor() const final;
+
+		/// <summary>
+		/// Gets the (GDI) device name (i.e. \\DISPLAY1) of the the monitor; do not use this as a persistent display identifier!
+		/// </summary>
+		/// <remarks>
+		/// This device name is not valid as a persistent display identifier for storage in configuration files, it may not refer to the same display device after a reboot.
+		/// </remarks>
+		virtual const wchar_t* get_device_name() const final;
```

Comment:
The ReShade API uses UTF-8 strings everywhere else, so ideally these should be converted too. I'm still in favor of a `reshade::api::device::get_property` alike instead of all the individual getter functions, but maybe that's just me =P